UnitsNet.Tests/BaseDimensionsTests.cs

```diff
@@ -0,0 +1,580 @@
+            Assert.True(result.LuminousIntensity == 0);
+        }
+
+        [Fact]
+        public void LuminousIntensityDimensionsDivideCorrectly()
+        {
+            var baseDimensions1 = new BaseDimensions(0, 0, 0, 0, 0, 0, 2);
+            var baseDimensions2 = new BaseDimensions(0, 0, 0, 0, 0, 0, 1);
+
+            var result = baseDimensions1.Divide(baseDimensions2);
+
+            Assert.True(result.Length == 0);
+            Assert.True(result.Mass == 0);
+            Assert.True(result.Time == 0);
+            Assert.True(result.Current == 0);
+            Assert.True(result.Temperature == 0);
+            Assert.True(result.Amount == 0);
+            Assert.True(result.LuminousIntensity == 1);
+        }
+
+        [Fact]
+        public void CheckBaseDimensionDivisionWithSpeedEqualsDistanceDividedByTimeOnStaticProperty()
+        {
+            var calculatedDimensions = Length.BaseDimensions.Divide(Duration.BaseDimensions);
+            Assert.True(calculatedDimensions == Speed.BaseDimensions);
+        }
+
+        [Fact]
+        public void CheckBaseDimensionDivisionWithSpeedEqualsDistanceDividedByTimeOnInstanceProperty()
+        {
+            var length = Length.FromKilometers(100);
+            var duration = Duration.FromHours(1);
+
+            var calculatedDimensions = length.Dimensions.Divide(duration.Dimensions);
+            Assert.True(calculatedDimensions == Speed.BaseDimensions);
+        }
+
+        [Fact]
+        public void CheckBaseDimensionMultiplicationWithForceEqualsMassTimesAccelerationOnStaticProperty()
+        {
+            var calculatedDimensions = Mass.BaseDimensions.Multiply(Acceleration.BaseDimensions);
+            Assert.True(calculatedDimensions == Force.BaseDimensions);
+        }
+
+        [Fact]
+        public void CheckBaseDimensionMultiplicationWithForceEqualsMassTimesAccelerationOnInstanceProperty()
+        {
+            var mass = Mass.FromPounds(205);
+            var acceleration = Acceleration.FromMetersPerSecondSquared(9.8);
+
+            var calculatedDimensions = mass.Dimensions.Multiply(acceleration.Dimensions);
+            Assert.True(calculatedDimensions == Force.BaseDimensions);
+        }
+
+#if !WINDOWS_UWP
+        [Fact]
+        public void EqualityWorksAsExpectedWithOperatorOverloads()
+        {
+            var baseDimensions1 = new BaseDimensions(1, 2, 3, 4, 5, 6, 7);
+            var baseDimensions2 = new BaseDimensions(1, 2, 3, 4, 5, 6, 7);
+
+            Assert.True(baseDimensions1 == baseDimensions2);
+            Assert.True(baseDimensions2 == baseDimensions1);
+        }
+
+        [Fact]
+        public void LengthDimensionsMultiplyCorrectlyWithOperatorOverloads()
+        {
+            var baseDimensions1 = new BaseDimensions(1, 0, 0, 0, 0, 0, 0);
+            var baseDimensions2 = new BaseDimensions(2, 0, 0, 0, 0, 0, 0);
+
+            var result = baseDimensions1 * baseDimensions2;
+
+            Assert.True(result.Length == 3);
```

Comment:
We don't have to do this now, but you could DRY this up a bit by adding a custom `AssertDimensionsEqual(BaseDimension d, int length, int mass, int time, int current, int temperature, int amount, int luminousIntensity)` method.

Reply:
Indeed we could. I could also test Equals with a new BaseDimensions. I think I like that better - you?